### ISSUE TITLE
Crashing when trying to connect to database (v .5)

### ISSUE THREAD

**Issue by netdawgx** [firebase/FirebaseUI-iOS#124]
I will update tomorrow with a properly submitted detailed report but for now here is an overview:

A project of mine broke after updating to this version. The project was able to be built but when trying to write to the database there was a crash. 

The error reported the Firebase default app was not configured. It also reported I was trying to access the database reference before the app was configured.

When I removed the pod the database was written to successfully, no crash.

**Comment by RobertBiehl** [comment]
I think I'm experiencing the same/similar issue.

**App crashes in line 2 of**
```
✅ FIRApp.configure()
💥 FIRDatabase.database().persistenceEnabled = true
```
with the following log
````
2016-09-11 12:04:29.828 Messenger[1190:389653] The default app has not been configured yet.
2016-09-11 12:04:29.829 Messenger[1190:389653] *** Terminating app due to uncaught exception 'FIRAppNotConfigured', reason: 'Failed to get default FIRDatabase instance. Must call FIRApp.configure() before using FIRDatabase.'
*** First throw call stack:
(0x25879b0b 0x25036dff 0x25879a51 0x189b4f 0xcdeec 0xcf5c0 0x29ecea6f 0x2a0f8287 0x2a0fc241 0x2a110811 0x2a0f9767 0x26f47bf7 0x26f47aa7 0x26f47da5 0x2583b9e7 0x2583b5d7 0x2583993f 0x257881c9 0x25787fbd 0x29ec7f37 0x29ec2435 0xd1294 0x25434873)
libc++abi.dylib: terminating with uncaught exception of type NSException
````

**On a sidenote, but might be related: **
Interestingly I'm using 0.5.2 and experiencing the duplicate framework linking like in 0.5.1. [Fix was supposed to be in 0.5.2](https://github.com/firebase/FirebaseUI-iOS/releases/tag/v0.5.2)
````
objc[1190]: Class FIRAAppEnvironmentUtil is implemented in both /private/var/containers/Bundle/Application/4241F344-EB64-4CCB-A9CF-1F2FD6222AC9/Messenger.app/Frameworks/FirebaseDatabaseUI.framework/FirebaseDatabaseUI and /private/var/containers/Bundle/Application/4241F344-EB64-4CCB-A9CF-1F2FD6222AC9/Messenger.app/Frameworks/FirebaseAuthUI.framework/FirebaseAuthUI. One of the two will be used. Which one is undefined.
objc[1190]: Class FIRAAppDelegateProxy is implemented in both /private/var/containers/Bundle/Application/4241F344-EB64-4CCB-A9CF-1F2FD6222AC9/Messenger.app/Frameworks/FirebaseDatabaseUI.framework/FirebaseDatabaseUI and /private/var/containers/Bundle/Application/4241F344-EB64-4CCB-A9CF-1F2FD6222AC9/Messenger.app/Frameworks/FirebaseAuthUI.framework/FirebaseAuthUI. One of the two will be used. Which one is undefined.
````

**Comment by RobertBiehl** [comment]
@netdawgx Can you please confirm that this is similar to what you are seeing? If not I will create a new issue.

**Comment by RobertBiehl** [comment]
See also #123

**Comment by netdawgx** [comment]
Yes, can confirm this is exactly what I'm seeing.

**Comment by RobertBiehl** [comment]
I now downgraded to 0.5.0 until a fix is available. 0.5.0 seems to have huge problems with iOS10. But that could also be a problem with Firebase itself.

**Comment by morganchen12** [comment]
Yeah, :(((( I broke this when converting to dynamic libs. Going to try to release a fixed version tomorrow. The issue is every dylib has a static dependency on Firebase, so they all link Firebase at compile time, but when loading the dylibs at runtime, since they each have their own Firebase, there's a bunch of duplicate FIRApp classes and they'll crash because only one of them ever gets properly initialized. 0.5.0 should be stable, but does not contain bitcode. Sorry all

**Comment by morganchen12** [comment]
This is fixed in 0.5.3.
